### ISSUE TITLE
[bug] function in array is detected as component and triggers `prefer-destructuring-assignment` lint

### ISSUE THREAD

**Issue by ulrichstark** [Rel1cx/eslint-react#1416]
### Describe the bug

I know detecting whether something is a React component or not is quite tricky.
But I now found a case where it's clearly not a React component, but still treated as one.
In the following reproduction, the function is an item of an array and therefore clearly not a React component.
Because it's treated as a React component, the lint rule `prefer-destructuring-assignment` flags my `data.supplierCompany`.

### Reproduction

```ts
type DeliveryNoteCheck = (data: { supplierCompany: string | null }) => string | null;

const deliveryNoteChecks: DeliveryNoteCheck[] = [
  (data) => {
    if (!data.supplierCompany) return null;
    return "Check for supplier company passed.";
  },
];
```

### Expected behavior

No lints triggered, because it's not relevant code for this plugin.

### Platform and versions

```Text
@eslint-react/eslint-plugin: 2.7.0
```

### Stack trace

```Text

```

### Additional context

Maybe just exclude function from the component detection logic if they are somewhere inside an array expression?
